frontend/src/slicer/customerSlice.ts

```diff
@@ -0,0 +1,158 @@
+    edit as apiEdit,
+    del as apiDelete
+} from '../services/apiService'
+import {emptyCustomer, ICustomer} from "../interfaces/ICustomer";
+import {hideDetails} from "./detailsSlice";
+import {
+    setPending,
+    stopPendingAndHandleError,
+    handleError,
+    invalidDataError,
+    handleApiResponse,
+} from "./errorHelper";
+
+
+const initialState: ICustomersState = {
+    customers: [],
+    current: emptyCustomer,
+    pending: false,
+    success: false,
+    toSave: emptyCustomer
+}
+const route = "orders_customers";
+
+export const validateCustomer = (customer: ICustomer): boolean => {
+    const necessaryValues = ['name', 'suppliers', 'purchasePrice', 'unitSize']
+    //@ts-ignore values defined in line above must be keys of ICustomer
+    if (necessaryValues.every(v => !!customer[v])) {
+        //@ts-ignore line above checks that values are not undefined
+        return customer.name.length > 0 && customer.suppliers.length > 0 && customer.purchasePrice > 0 && customer.unitSize > 0
+    }
+    return false
+}
+
+const validateBeforeSendingToBackend = ({customer}: RootState) => {
+    return validateCustomer(customer.toSave);
+}
+
+const hideDetailsAndReloadList = (dispatch: Dispatch) => {
+    dispatch(hideDetails());
+    //@ts-ignore
+    dispatch(getAllOpenCustomers());
+}
+
+export const getAllOpenCustomers = createAsyncThunk<IResponseGetAllCustomers, void, { state: RootState, dispatch: Dispatch }>(
+    'customers/getAll',
+    async (_, {getState, dispatch}) => {
+        const token = getState().authentication.token;
+        const {data, status, statusText} = await apiGetAll(route + '/all', token);
+        handleError(status, statusText, dispatch);
+        return {data, status, statusText}
+    }
+)
+
+export const getOneCustomer = createAsyncThunk<IResponseGetOneCustomer, string, { state: RootState, dispatch: Dispatch }>(
+    'customers/getOne',
+    async (id, {getState, dispatch}) => {
+        const token = getState().authentication.token
+        const {data, status, statusText} = await apiGetOne(route, token, id);
+        handleError(status, statusText, dispatch);
+        return {data, status, statusText}
+    }
+)
+
+export const createCustomer = createAsyncThunk<IResponseGetOneCustomer, void, { state: RootState, dispatch: Dispatch }>(
+    'customers/create',
+    async (_, {getState, dispatch}) => {
+        console.log(123)
+        const token = getState().authentication.token
+        const {data, status, statusText} = await apiCreate(route, token, getState().customer.toSave);
+        handleError(status, statusText, dispatch);
+        if (status === 200) hideDetailsAndReloadList(dispatch);
+        return {data, status, statusText}
+    }
+)
+
+export const editCustomer = createAsyncThunk<IResponseGetOneCustomer, ICustomer, { state: RootState, dispatch: Dispatch }>(
+    'customers/edit',
+    async (customer, {getState, dispatch}) => {
+        if (!validateBeforeSendingToBackend(getState())) {
```

Comment:
why not directly call `validateCustomer`?

Reply:
i need to be able to use the exported validate function in other parts of the application.
When I'm trying to call this one directly from the thunk like this 
` validateCustomer(getState().customer)`
it gives me an error. As it is it's working error free